app/window.py

```diff
@@ -578,7 +578,7 @@ def onChange(self):
           lineCursor -= 1
     pathLine = self.host.textBuffer.fullPath
     if 1:
-      if tb.isReadOnly:
+      if tb.fileStats.getUpdatedFileInfo()['isReadOnly']:
```

Comment:
This can be called frequently (when scrolling the window for example). Acquiring a semaphore and doing some file IO can be heavy. What happens if we use the saved (cached) value? It could be slightly out of date, but that won't last long. I think I'd prefer to have some lag on updating the RO state (happens infrequently) than lag when scrolling (happens frequently).

Reply:
Oh that sounds good. I did this because it also supported single threading nicely. Should be fixed now.